codegen/templates/templates.go

```diff
@@ -22,27 +22,105 @@ import (
 // this is done with a global because subtemplates currently get called in functions. Lets aim to remove this eventually.
 var CurrentImports *Imports
 
-func RenderToFile(tpl string, destFile string, data interface{}) error {
-	if tpl == "" {
-		return fmt.Errorf("no template name given")
-	}
+type Options struct {
+	PackageName     string
+	Filename        string
+	RegionTags      bool
+	GeneratedHeader bool
+	Data            interface{}
+}
+
+func Render(cfg Options) error {
 	if CurrentImports != nil {
 		panic(fmt.Errorf("recursive or concurrent call to RenderToFile detected"))
 	}
-	CurrentImports = &Imports{destDir: filepath.Dir(destFile)}
+	CurrentImports = &Imports{destDir: filepath.Dir(cfg.Filename)}
+
+	// load path relative to calling source file
+	_, callerFile, _, _ := runtime.Caller(1)
```

Comment:
Neat trick 😄 I was super confused about how templates were being referenced for a minute there. May eventually require docs since it will probably be something plugins want to commonly do (or a separate interface)